script/tool/lib/src/publish_plugin_command.dart

```diff
@@ -162,56 +360,67 @@ class PublishPluginCommand extends PluginCommand {
     return packageDir;
   }
 
-  Future<void> _checkGitStatus(Directory packageDir) async {
-    final ProcessResult statusResult = await processRunner.runAndExitOnError(
-        'git',
-        <String>[
-          'status',
-          '--porcelain',
-          '--ignored',
-          packageDir.absolute.path
-        ],
-        workingDir: packageDir);
+  Future<bool> _checkGitStatus(Directory packageDir) async {
+    io.ProcessResult statusResult;
+    try {
+      statusResult = await processRunner.runAndExitOnError(
```

Comment:
The naming here suggests there's a version that doesn't exit on error, so we wouldn't need to catch ToolExits?

Reply:
`runAndExitOnError` does something extra (printing error message in a specific format etc), if we don't use it, then we have to re-write that code. I think we should rename this method to `run(bool logError = true, bool exitOnError = true...)`. 

Actually, I just saw that the `run` method that we have already has a `exitOnError` flag which doesn't do anything, which suggests that we used to have it like this but somehow refactored into 2 methods. https://github.com/flutter/plugins/blob/master/script/tool/lib/src/common.dart#L507

I'm happy to do the refactoring in a separate PR! Then rebase this on top after that lands.

Reply:
https://github.com/flutter/plugins/pull/3827 @stuartmorgan here you go